docs/guide/server-events.md

```diff
@@ -0,0 +1,122 @@
+
+Every event includes particular arguments, so event handlers need to be written with matching parameters. For example, the `beforeInsert` event from piece type modules includes the request (`req`) and an object of data for the inserted piece (`piece`).
+
+The examples below show two examples of how a custom `product` module registers event handlers for that event.
+
+<AposCodeBlock>
+  ```javascript
+  module.exports = {
+    // ...
+    handlers(self) {
+      return {
+        // Responds to `beforeInsert` when emitted by the `product` module
+        beforeInsert: {
+          async applyTax(req, piece) {
+            piece.totalPrice = piece.price * (1.0 + (piece.tax / 100));
+          }
+        },
+        // Response to `beforeInsert` when emitted by *any* piece type
+        '@apostrophecms/piece-type:beforeInsert': {
+          async beforeAnyPieceIsInserted(req, piece) {
+            console.log('Something is being inserted. 📬', piece.title);
+          }
+        }
+      }
+    }
+  };
+  ```
+  <template v-slot:caption>
+    modules/product/index.js
+  </template>
+</AposCodeBlock>
+
+The first event handler object is set to watch `beforeInsert`. Since that key only includes the event name it will watch for the event **only when emitted by this `product` module**.
+
+The second event handler object is set to the key `'@apostrophecms/piece-type:beforeInsert'`. This still watches the `beforeInsert` event, but **since it begins with a module name**, `@apostrophecms/piece-type` followed by a colon, it will run the handler whenever that module, *or any modules that extend it*, emit that event. Since all piece types extend that named module, this handler will run before *any* piece is inserted into the database.
+
+## Multiple handlers for an event
+
+Event handlers can be spread *across multiple modules* in a project even if they respond to the same event. This allows us to keep each handler in the module that is most related to their purpose. When a product is updated by an editor, one module may send emails to notify the sales team and another module may update featured products on the home page.
+
+<AposCodeBlock>
+  ```javascript
+  module.exports = {
+    // ...
+    handlers(self) {
+      return {
+        // Responds to `beforeSave` when emitted by the `product` module
+        'product:beforeSave': {
+          async updateProducts(req, piece) {
+            // Code to update featured products for the home page
+          }
+        }
+      }
+    }
+  };
+  ```
+  <template v-slot:caption>
+    modules/@apostrophecms/home-page/index.js
+  </template>
+</AposCodeBlock>
+
+<AposCodeBlock>
+  ```javascript
+  module.exports = {
+    // ...
+    handlers(self) {
+      return {
+        // Responds to `beforeSave` when emitted by the `product` module
+        beforeSave: {
+          async emailSales(req, piece) {
+            // Code to construct and send an email to the sales team
```

Comment:
In many situations like this I think real-life code can distract from the point of the guide. If I added hacky, basic email notification logic here that adds several unrelated lines that someone may scrutinize rather than focusing on event handling.

Reply:
OK yes, I'm probably suggesting this because we just don't have enough guides yet for these topics.